qualang_tools/units/units.py

```diff
@@ -7,15 +7,68 @@
     - raw2volts: converts raw data to volts.
 """
 
+from qm.program._Program import _Program
+from inspect import stack
+from warnings import warn
+
+class _nanosecond:
 
-class unit:
     def __init__(self):
+        pass
+
+    def _get_value(self) -> float:
+
+        for frameinfo in stack():
+            
+            for var in frameinfo.frame.f_locals.values():
+
+                if isinstance(var, _Program): # we have a qua program being declared somewhere
+                    if var._is_in_scope: # this is set by __enter__ and unset by __exit__
+                        return 0.25 # now it is in clock cycles
+        else:
+            return 1.0 # this is in nanosecods
+
+    def __mul__(self, other : int | float ) -> float:

```

Comment:
Thank you so much for the comments @yomach !

I have run black and updated the type hints :)

I am completely in favor of making the casting to `int` the default with the verbose options. 
If this is the default behavior, maybe it would be safer for users to keep the warning but have it round up and make times longer, to avoid having short intervals become `0`?

I also had a go at adding a new `Time Units` section in units/README.md, happy to have feedback on that! :)

Reply:
I think that rounding it down is better because it is the usual way casting is done.
In addition, the minimum pulse duration is currently 4 cycles...

Reply:
I wrote in the review, but the changelog and readme are well written :)

Reply:
Thanks!

Reply:
> I think that rounding it down is better because it is the usual way casting is done. In addition, the minimum pulse duration is currently 4 cycles...

Okay! I did not remember this limitation. Then I also aggree with rounding down